ui/src/alerting/components/builder/CheckMatchingRulesCard.tsx

```diff
@@ -1,19 +1,86 @@
 // Libraries
-import React, {FunctionComponent} from 'react'
+import React, {FunctionComponent, useState, useEffect} from 'react'
+import {connect} from 'react-redux'
+import {get} from 'lodash'
 
 // Components
+import MatchingRuleCard from 'src/alerting/components/builder/MatchingRuleCard'
+
+// Actions
+import {getActiveTimeMachine} from 'src/timeMachine/selectors'
+
+// API
+import * as api from 'src/client'
+
+//Types
+import {NotificationRule, Check, AppState, CheckTagSet} from 'src/types'
 import {EmptyState, ComponentSize} from '@influxdata/clockface'
 
-const CheckMatchingRulesCard: FunctionComponent = () => {
+interface StateProps {
+  check: Partial<Check>
+  orgID: string
+}
+
+const CheckMatchingRulesCard: FunctionComponent<StateProps> = ({
+  check,
+  orgID,
+}) => {
+  const getMatchingRules = async (): Promise<NotificationRule[]> => {
+    const tags: CheckTagSet[] = get(check, 'tags', [])
+    const tagString = tags.map(t => `tag=${t.key}:${t.value}`).join('&')
```

Comment:
Check out the [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLsearchParams) API for doing this sort of thing... it's pretty nice and handles URL escaping and all that

Reply:
Actually I'm not sure this will work either way, [oats](https://github.com/chnn/oats) probably needs to be updated to handle serializing an array of values as a query parameters.

Different APIs encode array query parameters differently. Trying to assign the array `['foo', 'bar', 'baz']` to the query paramater `filter` sometimes results in
```
?filter=foo&query=bar&query=baz
```
or sometimes
```
?filter=foo,bar,baz
```
depending on the API. So I'm not sure what to support in oats, without breaking the generality of the generator.

I suppose the practical thing is just to support the former style, which seems to be what Influx uses. Or maybe we just write a custom API utility function—that's probably my preference. We can still use the generated request/response types from oats to ensure it stays in sync with the swagger.

Reply:
yup definitely doesn't work as is! let's take a look at this tomorrow?

Reply:
solved!